Assets/MRTK/SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -255,6 +260,14 @@ private void OnDisable()
         /// </summary>
         public void StartPlacement()
         {
+            // Check to see if Start() has been called, if not set AutoStart to true. This will make sure StartPlacement() will be
+            // called again when Start() is called.
+            if (!startCalled)
+            {
+                AutoStart = true;
```

Comment:
@CDiaz-MS what are your thoughts here?

Sometimes having an unrelated function modify other parameters under the cover (especially ones that people can set/control themselves) can be weird.

Consider the following in consuming code:

1. I call StartPlacement but tap to place hasn't actually started yet
2. I then set AutoStart to false (thinking, hey, I already told it to start placement, so I don't need future enable/disable transitions to auto start as well).

The difference here is that the StartPlacement call itself is really a one time thing (i.e. I want it to start once) but AutoStart is actually a many time thing (any time this starts, I want it to begin placement).

I think that for the purposes of this though, I *think* that what you have here is fine assuming we update the docs for both (i.e. and also guide people to use StartPlacement if they actually want to cause the one time auto start, and use AutoStart for the other case)

Reply:
I think overriding a public property that a user could have intentionally set to false (AutoStart) could potentially be confusing.

The following two scenarios could happen:

``tapToPlace.AutoStart = false``
``tapToPlace.StartPlacement()``
This works, but AutoStart is overwritten to true in StartPlacement and the user did intentionally set it to false.

``tapToPlace.StartPlacement()``
``tapToPlace.AutoStart = false``
Here the StartPlacement is unknowingly canceled out because AutoStart was set to false.  The user does not know that the timing of a StartPlacement call could set AutoStart to true when they specifically set it to false.

Instead of potentially overriding the value of a public property without the user knowing, what if we added a new private property to cover the specific scenario that StartPlacement is called before TapToPlace.Start.  This could disconnect the timing of a StartPlacement call to the value of AutoStart.

Reply:
@wiwei @CDiaz-MS I agree with you that manipulating a public property could be potentially dangerous, but I made the decision because I didn't want to add another private variable. Now that you both mentioned the current proposal could be problematic I will just go ahead and add a private variable. Thanks!